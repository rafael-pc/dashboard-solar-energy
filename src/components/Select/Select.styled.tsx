import styled from "styled-components";

export const Input = styled.select<{ errorMessage?: string }>`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 8px;
  border: 1px solid ${(props) => (props.errorMessage ? "#dc3545" : "#6e6e6e")};
  border-radius: 5px;
  width: 50%;
  height: 38px;
  margin-top: 10px;
  //color: #686868;
  background-color: #fff;
  outline: none;
  font-size: 1em;

  &:hover {
    box-shadow: ${(props) =>
      props.errorMessage
        ? "#dc3545"
        : "rgba(0, 0, 0, 0.16) 0px 1px 4px, rgb(51, 51, 51) 0px 0px 0px 1px"};
  }

  .icon {
    width: 70px;
    height: 30px;
    color: #a8a8a8;
  }

  @media only screen and (max-width: 600px) {
    height: 35px;
  }

  @media only screen and (max-width: 400px) {
    .envelope-icon {
      display: none;
    }
  }

  @media only screen and (max-height: 420px) {
    font-size: 0.9rem;
  }
`;

export const ErrorMessage = styled.div`
  color: #dc3545;
  font-size: 0.9em;
  padding: 6px;
  padding-left: 0;
  font-weight: bold;
`;
