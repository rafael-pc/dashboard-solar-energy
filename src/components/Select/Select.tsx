import { FC, SelectHTMLAttributes } from "react";
import { InputSelect } from "./Select.styled";

interface ISelectProps extends SelectHTMLAttributes<HTMLSelectElement> {
  props?: unknown;
}

const Select: FC<ISelectProps> = ({ ...props }) => {
  return <InputSelect {...props}/>;
};

export default Select;
