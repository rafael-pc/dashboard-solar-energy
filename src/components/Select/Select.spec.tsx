import { render, screen } from '@testing-library/react';
import Select from './Select';
import "@testing-library/jest-dom";

describe('Select', () => {
  it('renders select element', () => {
    const { asFragment } = render(<Select />);
    const selectElement = screen.getByRole('combobox');
    expect(selectElement).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders options', () => {
    const options = ['Option 1', 'Option 2'];
    render(<Select>{options.map(option => <option key={option}>{option}</option>)}</Select>);
    options.forEach(option => {
      expect(screen.getByText(option)).toBeInTheDocument();
    });
  });
});
