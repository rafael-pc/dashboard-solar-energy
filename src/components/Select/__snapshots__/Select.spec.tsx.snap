// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders select element 1`] = `
<DocumentFragment>
  <select
    class="sc-bcPKhP cesjdj"
  />
</DocumentFragment>
`;
